app/controllers/api/braintree_controller.rb

```diff
@@ -4,7 +4,26 @@ def token
   end
 
   def transaction
-    # we could use permitted params instead of params.except(:action, :controller)
-    render json: PaymentProcessor::Clients::Braintree::Transaction.make_transaction(params.except(:action, :controller))
+    sale = braintree.make_transaction(options)
+
+    if sale.success?
+      render json: { success: true, transaction_id: sale.transaction_id }
+    else
+      render json: { success: false, errors: [] }
+    end
+  end
```

Comment:
I'd probably keep this bit out of the controller and move it to the transaction class, since in reality we'll probably want more complicated logic for the return object. I'll see if I can sketch it out.